src/metabase/native_query_analyzer.clj

```diff
@@ -30,37 +29,49 @@
        (or (not config/is-test?)
            *parse-queries-in-test?*)))
 
-(defn- normalize-name
-  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
-  ;; c.f. Milestone 3 of the epic: https://github.com/metabase/metabase/issues/36911
-  [name]
-  (-> name
-      (str/replace "\"" "")
-      u/lower-case-en))
-
 (def ^:private field-and-table-fragment
   "HoneySQL fragment to get the Field and Table"
   {:from [[:metabase_field :f]]
    ;; (t2/table-name :model/Table) doesn't work on CI since models/table.clj hasn't been loaded
    :join [[:metabase_table :t] [:= :table_id :t.id]]})
 
+(defn- field-query
+  "Exact match for quoted fields, case-insensitive match for non-quoted fields"
+  [field value]
+  (if (= (first value) \")
+    [:= field (subs value 1 (dec (count value)))]
+    [:= [:lower field] (u/lower-case-en value)]))
+
+(defn- table-query
+  [t]
+  (if-not (:schema t)
+    (field-query :t.name (:table t))
+    [:and
+     (field-query :t.name (:table t))
+     (field-query :t.schema (:schema t))]))
```

Comment:
I could see somebody complaining about the overloaded use of `field-query`, but I think it's pragmatic here, so :+1: from me

Reply:
It's kind of... what can you do? We need those quotes handled and you could easily write `FROM "public".table`.